spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubProperties.java

```diff
@@ -47,6 +47,14 @@
 	@NestedConfigurationProperty
 	private final Credentials credentials = new Credentials(GcpScope.PUBSUB.getUrl());
 
+	/** The specific topic that should be used. Useful for environments like Cloud Foundry. */
+	private String topicName;
```

Comment:
How's this supposed to be used because it doesn't seem to be used by this project at all? Maybe we just need a better documentation, but this may confuse regular users.

Reply:
Well, that should be as good addition anyway.
We have similar approach for RabbitMQ and Kafka auto-configurations: https://github.com/spring-projects/spring-boot/blob/v2.0.1.RELEASE/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java#L763

Let's really address it ASAP after merging this PR!

Reply:
OK, I can kind of understand the purpose of a default topic, but then this field should probably be renamed to reflect that.